dist/privacy-sandbox.js

```diff
@@ -0,0 +1,194 @@
+    const isDocument = request.type === 'Document';
+    const cannonicalRequestDomain = getCanonicalDomain(url.hostname);
+
+    let thirdPartyDomain = '';
+    // Maps all first party hostnames to the corresponding cannonical first party domain to truly consider only third-parties
+    if (cannonicalRequestDomain !== cannonicalFirstPartyDomain) {
+      thirdPartyDomain = url.hostname;
+    }
+    if (thirdPartyDomain && seenThirdParties.includes(thirdPartyDomain)) {
+        continue;
+    }
+    seenThirdParties.push(thirdPartyDomain);
+
+    /**
+     * Topics API
+     * API Usage Reference: https://developers.google.com/privacy-sandbox/relevance/topics/demo#the-topics-api-demo
+     * Header Usage Reference: https://developers.google.com/privacy-sandbox/relevance/topics/demo#the-topics-api-demo
+     */
+
+    // If the request is to fetch a javascript file or HTML document, perform string search in the returned file content for usage of Topics API
+    // document.browsingTopics() [JS] or fetch() request call includes: {browsingTopics: true} or XHR request call includes: {deprecatedBrowsingTopics: true} (to be deprecated)
+    if (isScript || isDocument) {
+      if (request.response_body && (request.response_body.includes('browsingTopics') || request.response_body.includes('deprecatedBrowsingTopics'))) {
+        result['topicsAPI']['topicsAccessJs'].push(thirdPartyDomain);
+      }
+    }
+
+    // Checking request header usage of Topics: header: 'Sec-Browsing-Topics: true'
+    if ('sec-browsing-topics' in request.request_headers.toLowerCase()) {
+      result['topicsAPI']['topicsAccessHeader'].push(thirdPartyDomain);
+    }
+
+    // Checking if sent Topics in the request (either via JS or headers) are observed by the receiver using response header 'Observe-Browsing-Topics'
+    // If value is ?1 then they are observed else they are not observed
+    let respHeaders = new Map(Object.entries(request.response_headers).map(([key, value]) => [key.toLowerCase(), value]));
+    if (respHeaders.has('observe-browsing-topics') && respHeaders.get('observe-browsing-topics') === "?1") {
+      result['topicsAPI']['observingTopics'].push(thirdPartyDomain);
+    }
+
+    const attestationPublished = await fetchAndCheckAttestation(`${url.origin}/.well-known/privacy-sandbox-attestations.json`);
+    if (attestationPublished) {
+      result.attestationPublished.push(thirdPartyDomain);
+    }
+
+    /**
+     * Protected Audience API
+     * https://github.com/WICG/turtledove/blob/main/FLEDGE.md
+     * https://developers.google.com/privacy-sandbox/relevance/protected-audience
+     */
+
+    // Checking if the request header includes 'Sec-Interest-Group-Storage' to join, leave or update an interest group
+    if (isScript || isDocument) {
+      if (request.response_body && (request.response_body.includes('joinAdInterestGroup('))) {
+        result['protectedAudienceAPI']['interestGroups']['joinAdInterestGroup'].push(cannonicalRequestDomain);
```

Comment:
Why do you use `cannonicalRequestDomain` here, but `thirdPartyDomain` for topics API?